plugin/makeshift.vim

```diff
@@ -87,6 +88,12 @@ function! s:makeshift()
     if len(l:program) == 0
         let l:program = s:determine_build_system(expand('%:p:h'))
     endif
+    if l:program == 'waf'
+        let l:found = globpath(b:makeshift_root, l:program)
+        if filereadable(l:found)
+            let l:program = l:found
+        endif
+    endif
```

Comment:
I've given this some more thought and can see the benefits of finding (e.g) `/path/to/project/bin/rake` for a Ruby project that bundles `rake`. I can also see a use-case where somebody working on a large C++ project might want to disable the search for `make`.

So... I _think_ (to answer my own questions) what we want here is something like (untested)...


```diff
diff --git a/plugin/makeshift.vim b/plugin/makeshift.vim
index c5ba840..ca5a59e 100644
--- a/plugin/makeshift.vim
+++ b/plugin/makeshift.vim
@@ -58,9 +58,21 @@ function! s:determine_build_system(dir)
     return ''
 endfunction
 
+function! s:find_bundled(program)
+    let l:bundled = globpath(b:makeshift_root, a:program)
+    if filereadable(l:bundled)
+        return l:bundled
+    endif
+    return a:program
+endfunction
+
 function! s:set_makeprg(program)
     if len(a:program)
-        let &l:makeprg=a:program
+        if exists('g:makeshift_find_bundled') && g:makeshift_find_bundled
+            let &l:makeprg = s:find_bundled(a:program)
+        else
+            let &l:makeprg = a:program
+        endif
     endif
 endfunction
 
@@ -88,12 +100,6 @@ function! s:makeshift()
     if len(l:program) == 0
         let l:program = s:determine_build_system(expand('%:p:h'))
     endif
-    if l:program == 'waf'
-        let l:found = globpath(b:makeshift_root, l:program)
-        if filereadable(l:found)
-            let l:program = l:found
-        endif
-    endif
     call s:set_makeprg(l:program)
     call s:set_makedir(len(l:program) > 0)
 endfunction
```

I don't have the capacity to test or document this right now. Do you want to have a go at this, @ieei?